src/components/[guild]/Requirements/components/Requirement.tsx

```diff
@@ -40,11 +42,35 @@ const Requirement = ({
   showViewOriginal,
 }: RequirementProps): JSX.Element => {
   const requirement = useRequirementContext()
+  const { setValue } = useFormContext() ?? {}
 
   const ChildrenWrapper = childrenWrapper ?? Box
+  const childrenWrapperProps = !!childrenWrapper
```

Comment:
These should be passed in RequirementEditableCard & ExistingRequirementEditableCard, ideally already injected into the Children- and ImageWrapper components with higher order components, so we don't need extra props for them. But actually I don't see any conditional logic / cases where RequirementImageEditor and RequirementNameAndVisibilityEditor works differently, so why not just abstract these setValues in them? As the other case, useEditRequirement is already abstracted in them. We should either extract that too and do what I've written in the first sentence, passing useEditRequirement in ExistingRequirementEditableCard and setValues in RequirementEditableCard (that would be proper composition, but would require an extra isLoading prop if I'm correct), or abstract these back too imo

Reply:
I went with the proper solution instead of abstracting back the logic, I feel like its better this way, but lmk 👍  e250cf8e2382b1877674dae6aa72d810e422085a